### DISCUSSION TITLE
More flexible object filtering for resolution on SnowDDL invocations via CLI interface

### DISCUSSION THREAD

**Discussion by acaruso7** [littleK0i/SnowDDL#90] [Ideas]
Currently, the CLI interface supports excluding or filtering for only specific object types during resolution via the `--exclude-object-types` and `--include-object-types` CLI flags, respectively. This is pretty useful, but if this functionality were extended to handle arbitrary selection of _specific named objects_, I think it could be really powerful.

For example, instead of using `--include-object-types DATABASE` to resolve only database objects, suppose a more granular selection method was available for just a single database that already exists in Snowflake. For instance, `--include-objects DATABASE:MY_RAW_DATABASE`. This way, if one wanted to execute a more targeted invocation, there's no risk of altering other objects which might contain diffs between the resource config and Snowflake state

This idea could be extended even further to target objects which are referenced in _any_ DDL expressions which appear in the plan, even if those objects are not the direct targets of the resolution. For example, suppose the plan is outputting the following change to a pre-existing role:

```
REVOKE READ ON FUTURE TABLES IN SCHEMA MY_RAW_DATABASE.MY_SCHEMA FROM ROLE MY_RAW_DATABASE__MY_SCHEMA__READ__S_ROLE;
```

A user may want to target or exclude this expression explicitly via specification of the individual objects referenced in the expression. For example, suppose one wanted to target grants which affect a specific database or specific schema:

```
--include-referenced-objects DATABASE:MY_RAW_DATABASE
```

or

```
--include-referenced-objects SCHEMA:MY_RAW_DATABASE.MY_SCHEMA
```

^^ Even though the resolved object for this expression would be `ROLE`, since it is _referencing_ both a `DATABASE` and `SCHEMA` in the expression, it might still be useful to target indirectly via those object references

**Comment by littleK0i** [comment]
I'll come back to this in a few days when I have a bit more time, please stay tuned.
